packages/components/app/styles/components/form/file-input.scss

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+//
+// FORM > FILE-INPUT
+//
+
+@use "../../mixins/button" as *;
+
+.hds-form-file-input {
+  margin: -4px 0 -4px -4px; // offset shadow padding + default 1px spacing
+  padding: 3px 0 3px 3px; // add padding so button shadow is visible
+  color: var(--token-color-foreground-primary);
+  border: 1px solid transparent;
+  border-radius: calc($hds-button-border-radius + 1px);
+
+  &:focus,
+  &:focus-visible {
+    outline: none;
+  }
+}
+
+// https://developer.mozilla.org/en-US/docs/Web/CSS/::file-selector-button
+// https://caniuse.com/?search=file-selector-button
+
+// Button
+.hds-form-file-input::file-selector-button {
+  min-height: 2.25rem;
+  margin-right: 16px;
+  margin-inline-end: 16px;
+  padding: 0.5625rem 0.9375rem;
+  padding-left: 34px; // override to add space for the background image icon
+  color: var(--token-color-foreground-primary);
+  font: inherit;
+  line-height: 1rem;
+  text-decoration: none;
```

Comment:
this is not needed, it will never be a link (this comes from the copy&paste of the `Hds::Button` styling)